Sources/Basics/FileSystem/Path+Extensions.swift

```diff
@@ -0,0 +1,41 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import struct TSCBasic.AbsolutePath
+
+extension AbsolutePath {
+    public func appending(_ component: String) -> AbsolutePath {
+        self.appending(component: component)
+    }
+
+    public func appending(_ components: String...) -> AbsolutePath {
+        self.appending(components: components)
+    }
+
+    public func withExtension(_ extension: String) -> AbsolutePath {
```

Comment:
Would be good if all appending-style APIs ended up grouped in auto-complete, so `appending(extension:)`? If we really want to mirror the "without" APIs, `with(extension:)`

Reply:
I like that!